src/test/scala/org/broadinstitute/dsde/firecloud/dataaccess/MockThurloeDAO.scala

```diff
@@ -134,7 +134,11 @@ class MockThurloeDAO extends ThurloeDAO {
   }
 
   override def saveKeyValues(userInfo: UserInfo, keyValues: Map[String, String]): Future[Try[Unit]] = {
-    val newKVsForUser = (userInfo.id -> (mockKeyValues(userInfo.id) ++ keyValues.map { case (key, value) => FireCloudKeyValue(Option(key), Option(value))}))
+    val newKVsForUser = userInfo.id -> (mockKeyValues(userInfo.id).filter {
+      case FireCloudKeyValue(Some(key), _) => !keyValues.contains(key)
+      case FireCloudKeyValue(_, _) => false
```

Comment:
good catch.. I spent a while trying to understand why we need this -- we need to join 2 sets of type ` Set[FireCloudKeyValue]` and we can't automatically replace key-value pairs because they're in the `FireCloudKeyValue` case class and not in a standard data structure like a Map.